WPMedia/fusion-news-theme-blocks
# Tmedia 724 header nav

## Description

Converted the header nav chain to use Advanced styling and an attempt to update it to new design:  https://www.figma.com/file/vWVGgZfZVBdSX7ACo3kmjr/Enhanced-Styling-Design-System?node-id=3507%3A35707

Corresponding feature PR:  https://github.com/WPMedia/arc-themes-feature-pack/pull/286

## Issue:

There is one test in that I comment out in `blocks/header-nav-chain-block/chains/header-nav-chain-block/default.test.jsx`.  I don't know if we can just remove it or need to fix it, but I don't understand it and can't fix it.

## Jira Ticket

- [[TMEDIA-](https://arcpublishing.atlassian.net/browse/TMEDIA-)](https://arcpublishing.atlassian.net/browse/TMEDIA-724)

## Acceptance Criteria

Header Nav Chain block is migrated to the new architecture
Includes Custom fields of:
‘Toggle light mode’ - defaulted to false
‘Chain Type' - with dropdown to select 'header-hav-chain-block”
'Arial-label' - open field with default of “Sections Menu” and existing help text
'Links Bar - Aria-label - open field with default of “Top Links” and existing help text
Includes Configure content fields of:
‘Sections Menu hierarchy’ - default is an empty field
‘Horizontal Links hierarchy' - default is an empty field
Includes Logo fields of:
‘Logo alignment' - Dropdown with options of “---”, “center” and “left”.  Defaults to center.
‘Starting desktop navigation bar height’ - open field with default pf 56.  Includes existing help text.
‘Shrink navigation bar after scrolling’ - default is an empty field. Includes existing help text.
Includes Display fields of:
‘Display dots between horizontal links' - checkbox, defaults to true (checked)
Includes Mobile Components of:
‘Left Component 1’ - Mobile - dropdown, “Sections Menu” is default
‘If custom, position of Left Component 1 - Mobile' - default is an empty field
‘Right Component 1 - Mobile’ - dropdown, defaults to “None”
‘If custom, position of Right Component 1 - Mobile’ - default is an empty field
‘Sections Menu Component 1 - Mobile’ - dropdown, defaults to “Arc Search”
‘If custom, position of Sections Menu 1 Component - Mobile’ - default is an empty field
‘Sections Menu Component 2 - Mobile’ - dropdown, defaults to “None”
‘If custom, position of Sections Menu Component 2 - Mobile’
Includes Desktop Components of:
‘Left Component 1’ - Desktop - dropdown, “Arc Search” is default
‘If custom, position of Left Component 1 - Desktop' - default is an empty field
‘Left Component 2 - Desktop’ - dropdown, defaults to “Sections Menu”
‘If custom, position of Left Component 2- Desktop’ - default is an empty field
‘Right Component 1 - Desktop’ - dropdown, defaults to “None”
‘If custom, position of Right Component 1 - Desktop’ - default is an empty field
‘Right Component 2 - Desktop’ - dropdown, defaults to “None”
‘If custom, position of Right Component 1 - Desktop’ - default is an empty field
‘Sections Menu Component 1 - Desktop - dropdown, defaults to “Arc Search”
‘If custom, position of Sections Menu 1 Component - Desktop’ - default is an empty field
‘Sections Menu Component 2 - Desktop’ - dropdown, defaults to “None”
‘If custom, position of Sections Menu Component 2 - Desktop’
Remove Dark/Light mode logic.

## Test Steps

First, run strobook on this branch for an initial test.  You can compare and contrast with an older storybook here: https://5f91e4721ccaba0022a10782-iseadguotq.chromatic.com/?path=/story/chains-header-nav-block--wide-logo. Just be advised that the new stories are using the updated design: https://www.figma.com/file/vWVGgZfZVBdSX7ACo3kmjr/Enhanced-Styling-Design-System?node-id=3507%3A35707

1. Checkout this branch `git checkout TMEDIA-724-header-nav`
2. Use this feature pack branch: `https://github.com/WPMedia/arc-themes-feature-pack/pull/286`
3. Ensure your .env is pointing to `FUSION_RELEASE=3.3.1`
4. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/header-nav-chain-block`

## Dependencies or Side Effects

_Examples of dependencies or side effects are:_

- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Author Checklist

_The author of the PR should fill out the following sections to ensure this PR is ready for review._

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._
